docs/support/troubleshooting/performance-troubleshooting.md

```diff
@@ -0,0 +1,242 @@
+
+This is a temporary solution. And will be reverted once the computer restarts.
+
+To make it permanent either,
+
+- Create a service to run this on startup (recommended)
+- **OR** put following lines to the `~/.bashrc` file:
+
+  ```bash
+  if [ ! -e /tmp/multicast_is_set ]; then
+  sudo ip link set lo multicast on
+  touch /tmp/multicast_is_set
+  fi
+  ```
+
+  - This will probably ask for password on the terminal every time you restart the computer.
+
+### Check the compilation flags
+
+#### Target symptoms
+
+- Autoware is running slower than expected
+- Point clouds are lagging
+- When multiple subscribers use the same publishers, the message rate drops even further
+
+#### Diagnosis
+
+Check the `~/.bash_history` file to see if there are any `colcon build` directives without `-DCMAKE_BUILD_TYPE=Release` or `-DCMAKE_BUILD_TYPE=RelWithDebInfo` flags at all.
+
+Even if a build starts with these flags but same workspace gets compiled without these flags, it will still be a slow build in the end.
+
+In addition, the nodes will run slow in general, especially the `pointcloud_preprocessor` nodes.
+
+Example issue: [issue2597](https://github.com/autowarefoundation/autoware.universe/issues/2597#issuecomment-1491789081)
+
+#### Solution
+
+- Remove the `build`, `install` and optionally `log` folders in the main `autoware` folder.
+- Compile the Autoware with either `Release` or `RelWithDebInfo` tags:
+
+  ```bash
+  colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
+  # Or build with debug flags too (comparable performance but you can debug too)
+  colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo
+  ```
+
+### Check the DDS settings
+
+#### Target symptoms
+
+- Autoware is running slower than expected
+- Messages show up late in RViz2
+- Point clouds are lagging
+- Camera images are lagging behind
+- When multiple subscribers use the same publishers, the message rate drops
+
+#### Check the RMW (ROS Middleware) implementation
+
+##### Diagnosis
+
+Run following to check the middleware used:
+
+```bash
+echo $RMW_IMPLEMENTATION
+```
+
+The return line should be `rmw_cyclonedds_cpp`. If not, apply the solution.
+
+If you are using a different DDS middleware, we might not have official support for it just yet.
+
+##### Solution
+
+Add `export RMW_IMPLEMENTATION=rmw_cyclonedds_cpp` as a separate line in you `~/.bashrc` file.
+
+#### Check if the CycloneDDS is configured correctly
+
+##### Diagnosis
+
+Run following to check the configuration `.xml` file of the `CycloneDDS`:
+
+```bash
+echo $CYCLONEDDS_URI
+```
+
+The return line should be a valid path pointing to an `.xml` file with `CycloneDDS` configuration.
+
+Also check if the file is configured correctly:
+
+```bash
+cat !{echo $CYCLONEDDS_URI}
+```
+
+This should print the `.xml` file on the terminal.
+
+##### Solution
+
+Follow [DDS settings](../../installation/additional-settings-for-developers/index.md#tuning-dds)
```

Comment:
The context of example config.xml seems different from the one in the linked paged.

Maybe we might want to delete the one in here and update the one in "Additional Settings for Developers" or the other way around so that we don't have to maintain the similar content in different places.

Reply:
Ok, I will create a PR to update the one in the link, and just keep the link here.

Reply:
Created https://github.com/autowarefoundation/autoware-documentation/pull/373 and should be merged before this pr.

Reply:
Fixed in https://github.com/autowarefoundation/autoware-documentation/pull/365/commits/d5e52ee553e580e62837727c3ef7d02830c80ca5

Actually they can be merged in any order you prefer, it doesn't matter.